tests/unit/test_e2e.py

```diff
@@ -849,10 +859,18 @@ def test_provider_get_maintenances(
     for result_parse_file in result_parse_files:
         with open(result_parse_file, encoding="utf-8") as res_file:
             partial_result_data = json.load(res_file)
-            if not expected_result:
+
+            # TODO: Tests assume that maintenances (multiple) will be discovered on the first parser
+            if not expected_result and isinstance(partial_result_data, list):
                 expected_result = partial_result_data
+
+            if expected_result and isinstance(partial_result_data, dict):
+                for _ in range(len(expected_result)):
+                    expected_result[0].update(partial_result_data)
             else:
-                expected_result[0].update(partial_result_data[0])
+                assert len(expected_result) == len(expected_result)
```

Comment:
Would this not always be true?

Reply:
if the expected results are well defined, yes :)

Reply:
you're right, typo there